modules/core/src/main/scala/sup/data/HealthReporter.scala

```diff
@@ -24,10 +24,25 @@ object HealthReporter {
   def wrapChecks[F[_]: Apply, G[_]: NonEmptyTraverse, H[_]: Reducible](checks: G[HealthCheck[F, H]])(
     implicit M: Semigroup[Health]): HealthReporter[F, G, H] = HealthCheck.liftF {
 
-    checks.nonEmptyTraverse(_.check).map { results =>
-      val status = results.reduceMap(_.value.reduce)
+    checks.nonEmptyTraverse(_.check).map(reduceResults[G, H] _)
+  }
+
+    /**
+    * Constructs a healthcheck from a non-empty structure G of healthchecks that are run in parallel.
+    * The status of the whole check is determined using the results of given checks and the semigroup of Health.
+    *
+    * e.g. if all checks need to return Healthy for the whole thing to be healthy, use [[Health.allHealthyCommutativeMonoid]].
+    */
+  def parWrapChecks[F[_]: Apply, G[_]: NonEmptyTraverse, H[_]: Reducible, M[_]](checks: G[HealthCheck[F, H]])(
+    implicit M: Semigroup[Health], P: Parallel[F, M]): HealthReporter[F, G, H] = HealthCheck.liftF {
+
+    Parallel.parNonEmptyTraverse(checks)(_.check).map(reduceResults[G, H] _)
```

Comment:
We could use syntax for parNonEmptyTraverse, but it's okay either way

Reply:
couldn't make it compile... or it doesn't exist or compiler is not getting some implicit.

Reply:
I'll handle it :)